tests/js/dictionary-tests.js

```diff
@@ -461,9 +460,43 @@ module.exports = {
 
         let error = new Error("The key: unknown_key doesn't exist in the Dictionary.")
         TestCase.assertThrowsException(() =>  realm.write(()=> {  D.remove( ['unknown_key'] )  }) , error)
-    }
-
-
+    },
+    testDictionaryWithNestedModels() {
+        const Child = {
+        name: "Children",
+            properties: {
+                num: "int"
+            }
+        }
+        
+        const DictSchema = {
+        name: "Dictionary",
+            properties: {
+                dict: "{}",
+                dict2: "{}"
+            }
+        }
+        
+        let realm = new Realm({schema: [DictSchema, Child]})
+        
+        realm.write(()=> {
+                    let child = realm.create(Child.name, { num: 555 } )
+                    let child2 = realm.create(Child.name, { num: 666 } )
+                    realm.create(DictSchema.name, {
+                                 dict: { children1: child, children2: child2 },
+                                 dict2: { children1: child2, children2: child }
+                                 })
+                    })
+        
+        let l = realm.objects(DictSchema.name)[0].dict
```

Comment:
The test seems to be incomplete. What happens when you uncomments the lines below?